backend/lib/security/index.js

```diff
@@ -162,9 +155,17 @@ async function authorizationUrl (req, res) {
   const redirectPath = frontendRedirectUrl.pathname + frontendRedirectUrl.search
   const redirectOrigin = frontendRedirectUrl.origin
   const backendRedirectUri = getBackendRedirectUri(redirectOrigin)
-  const state = encodeState({
+  const state = generators.state()
+  res.cookie(COOKIE_STATE, {
     redirectPath,
-    redirectOrigin
+    redirectOrigin,
+    state
+  }, {
+    secure: true,
+    httpOnly: true,
+    maxAge: 180_000, // cookie will be removed after 3 minutes
+    sameSite: 'Lax',
+    signed: true
```

Comment:
`signed` rm? except we know of a concrete attack vectore that we want to protect against